pkg/kubelet/kubelet.go

```diff
@@ -319,7 +319,10 @@ func NewMainKubelet(
 		return nil, err
 	}
 
-	procFs := procfs.NewProcFs()
+	procFs := procfs.NewProcFS()
+	if err := procfs.SetupRestartOnPanicAndOops(procFs); err != nil {
```

Comment:
This should be moved to within [container manager](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/container_manager_linux.go#L146).

Once that is done you will no longer have to inject a procFS interface.